### PR TITLE
add method: getUserThreadPoolMap

### PR DESCRIPTION
**Pull request by chuailiwu**
### Motivation:
add getUserThreadPoolMap method, user for monitoring it

### CODE REVIEW

File: core/api/src/main/java/com/alipay/sofa/rpc/config/UserThreadPoolManager.java [hunk 1]
```diff
@@ -76,4 +78,15 @@ public static UserThreadPool getUserThread(String service) {
         return userThreadMap == null ? null : userThreadMap.get(service);
     }
 
+    public static Map<String, UserThreadPool> getUserThreadPoolMap() {
```
Comment (nobodyiam) [inline core/api/src/main/java/com/alipay/sofa/rpc/config/UserThreadPoolManager.java:81]: Where is this method used? Is it a read-only scenario?

**Review by EvenLjj — APPROVED** [review]
LGTM

File: core/api/src/main/java/com/alipay/sofa/rpc/config/UserThreadPoolManager.java (hunk 1 above)
Comment (chuailiwu) [reply core/api/src/main/java/com/alipay/sofa/rpc/config/UserThreadPoolManager.java:81]: Right. It is used to monitor userThreadPool(currently only support monitoring default threadpool).

Comment (nobodyiam) [reply core/api/src/main/java/com/alipay/sofa/rpc/config/UserThreadPoolManager.java:81]: Is hashcode meaningful to the user? or we could implement some unique identifier mechanism, e.g. if the same name is already present, then we append a counter after it?
> SofaUserProcessor-1, SofaUserProcessor-2

File: core/api/src/test/java/com/alipay/sofa/rpc/config/UserThreadPoolManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+public class UserThreadPoolManagerTest {
+    @Test
+    public void getUserThreadPoolMap() {
+
+        Set<UserThreadPool> userThreadPoolSet = UserThreadPoolManager.getUserThreadPoolSet();
+        Assert.assertTrue(userThreadPoolSet.isEmpty());
+
+        UserThreadPool pool1 = new UserThreadPool();
+        Assert.assertEquals(pool1.getThreadPoolName(), UserThreadPool.DEFAUT_POOL_NAME + "-0");
```
Comment (nobodyiam) [inline core/api/src/test/java/com/alipay/sofa/rpc/config/UserThreadPoolManagerTest.java:32]: Shall we reset the POOL_NAME_COUNTER every time before we run this test? It looks like this test is not stable as UserThreadPool might be initialized in other unit tests.

**Review by nobodyiam — COMMENTED** [review]
It looks to me okay now, except the potential unit test issue below.

File: core/api/src/test/java/com/alipay/sofa/rpc/config/UserThreadPoolManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        UserThreadPool pool1 = new UserThreadPool();
+        Assert.assertTrue(pool1.getThreadPoolName().startsWith(UserThreadPool.DEFAUT_POOL_NAME + "-"));
+
+        UserThreadPoolManager.registerUserThread("service1", pool1);
+        userThreadPoolSet = UserThreadPoolManager.getUserThreadPoolSet();
+        Assert.assertTrue(userThreadPoolSet.size() == 1);
+
+        UserThreadPool pool2 = new UserThreadPool();
+        Assert.assertTrue(pool2.getThreadPoolName().startsWith(UserThreadPool.DEFAUT_POOL_NAME + "-"));
```
Comment (nobodyiam) [inline core/api/src/test/java/com/alipay/sofa/rpc/config/UserThreadPoolManagerTest.java:39]: I think we could also test `pool2.getThreadPoolName()` not equals to `pool1.getThreadPoolName()`.

**Review by nobodyiam — APPROVED** [review]
LGTM

**Review by OrezzerO — APPROVED** [review]
LGTM

**Pull request merged**